extensions/packageManager/src/client/Library.tsx

```diff
@@ -44,21 +55,36 @@ const Library: React.FC = () => {
   const [isLoadingInstalled, setIsLoadingInstalled] = useState<boolean>(false);
   const [recentlyUsed, setRecentlyUsed] = useState<LibraryRef[]>([]);
   const [runtimeLanguage, setRuntimeLanguage] = useState<string>('c#');
-  const [feeds, updateFeeds] = useState([]);
+  const [feeds, updateFeeds] = useState<PackageSourceFeed[]>([]);
   const [feed, setFeed] = useState<string | undefined>(undefined);
+  const [searchTerm, setSearchTerm] = useState<string>('');
   const [loading, setLoading] = useState(false);
   const [selectedItem, setSelectedItem] = useState<LibraryRef>();
+  const [selectedItemVersions, setSelectedItemVersions] = useState<string[]>([]);
+  const [selectedVersion, setSelectedVersion] = useState<string>('');
   const [currentProjectId, setCurrentProjectId] = useState<string>(projectId);
   const [working, setWorking] = useState(false);
   const [addDialogHidden, setAddDialogHidden] = useState(true);
+  const [isModalVisible, setModalVisible] = useState<boolean>(false);
+  const [readmeContent, setReadmeContent] = useState<string>('');
+  const [versionOptions, setVersionOptions] = useState<IContextualMenuProps|undefined>(undefined);
+  const [isUpdate, setIsUpdate] = useState<boolean>(false);
   const httpClient = useHttpClient();
   const API_ROOT = '';
-
+  const TABS = {
+    INSTALL: 'INSTALL',
+    BROWSE: 'BROWSE",'
+  };
+  const [currentTab, setCurrentTab] = useState<string>(TABS.BROWSE);
   const strings = {
```

Comment:
nit: to keep your component code more focused on it's logic, you can move this outside the component body and make it a function.